src/coreclr/jit/fgprofile.cpp

```diff
@@ -4257,265 +4257,17 @@ bool Compiler::fgIncorporateEdgeCounts()
-            // is more than newWeight, so we just allow for the slop
-            if ((newWeight + slop) >= m_edgeWeightMin)
-            {
-                result   = true;
-                usedSlop = true;
-
-                if (m_edgeWeightMax != BB_ZERO_WEIGHT)
-                {
-                    // We will allow this to lower m_edgeWeightMin and Max towards newWeight
-                    m_edgeWeightMax = m_edgeWeightMin;
-                    m_edgeWeightMin = newWeight;
-                }
-            }
-        }
-
-        // If we are returning true then we should have adjusted the range so that
-        // the newWeight is in new range [Min..Max] or fgEdgeWeightMax is zero
-        if (result)
-        {
-            assert((m_edgeWeightMax == BB_ZERO_WEIGHT) ||
-                   ((newWeight <= m_edgeWeightMax) && (newWeight >= m_edgeWeightMin)));
-        }
-    }
-
-    if (result && usedSlop && (wbUsedSlop != nullptr))
-    {
-        *wbUsedSlop = true;
-    }
-
-#if DEBUG
-    if (result)
-    {
-        JITDUMP("Updated max weight of " FMT_BB " -> " FMT_BB " to [" FMT_WT ".." FMT_WT "]\n", getSourceBlock()->bbNum,
-                bDst->bbNum, m_edgeWeightMin, m_edgeWeightMax);
-    }
-    else
-    {
-        JITDUMP("Not adjusting max weight of " FMT_BB " -> " FMT_BB "; new value " FMT_WT " not in range [" FMT_WT
-                ".." FMT_WT "] (+/- " FMT_WT ")\n",
-                getSourceBlock()->bbNum, bDst->bbNum, newWeight, m_edgeWeightMin, m_edgeWeightMax, slop);
-        result = false; // break here
-    }
-#endif // DEBUG
-
-    return result;
-}
-
-//------------------------------------------------------------------------
-// setEdgeWeights: Sets the minimum lower (m_edgeWeightMin) value
-//                  and the maximum upper (m_edgeWeightMax) value
-//                 Asserts that the max value is greater or equal to the min value
-//
-// Arguments:
-//    theMinWeight - the new minimum lower (m_edgeWeightMin)
-//    theMaxWeight - the new maximum upper (m_edgeWeightMin)
-//    bDst         - the destination block for the edge
-//
-void FlowEdge::setEdgeWeights(weight_t theMinWeight, weight_t theMaxWeight, BasicBlock* bDst)
-{
-    assert(theMinWeight <= theMaxWeight);
-    assert(theMinWeight >= 0.0);
-    assert(theMaxWeight >= 0.0);
-
-    JITDUMP("Setting edge weights for " FMT_BB " -> " FMT_BB " to [" FMT_WT " .. " FMT_WT "]\n",
-            getSourceBlock()->bbNum, bDst->bbNum, theMinWeight, theMaxWeight);
-
-    m_edgeWeightMin = theMinWeight;
-    m_edgeWeightMax = theMaxWeight;
-}
-
 //-------------------------------------------------------------
-// fgComputeBlockAndEdgeWeights: determine weights for blocks
-//   and optionally for edges
+// fgComputeBlockWeights: determine weights for blocks
 //
 // Returns:
 //    Suitable phase status
 //
-PhaseStatus Compiler::fgComputeBlockAndEdgeWeights()
+PhaseStatus Compiler::fgComputeBlockWeights()
```

Comment:
This phase should eventually go away. We can compute `fgCalledCount` much earlier now. But let's leave that for follow-up too.

Reply:
Are you suggesting we compute `fgCalledCount` in `fgIncorporateProfileData`? And would we still want to call `fgComputeMissingBlockWeights`? I tried removing this phase altogether locally and computing `fgCalledCount` during profile incorporation, and the diffs from this are quite big

Reply:
More or less -- but both may still be needed for non-pgo cases?

We are going to need to find the right reconciliation between having PGO and not having it. Ideally perhaps we just always think we have it, but I don't know if we want to pay the costs of this in modes like minopt where we probably don't ever look at the data...